### ISSUE TITLE
feature request: anonymous users

### ISSUE THREAD

**Issue by nahtnam** [logto-io/logto#5207]
### What problem did you meet?

I would love for potential customers to be able to create an anonymous account (no email address/username/password) and try out my app. Later they can add their email and verify it

### Describe what you'd like Logto to have

The ability to create an account with just a name (no way to sign in back in, that's okay). Then later add an email address to finish setting their account up. Firebase supports this with anonymous users

**Comment by charIeszhao** [comment]
Hi @nahtnam, thanks for reaching out. This is an interesting feature, however it is currently not on our radar.

And what I don't quite understand is how you can tell the difference between an anonymous user and a unregistered user? Seems to me that an anonymous user is almost identical to not having a user account at all, right?

**Comment by nahtnam** [comment]
@charIeszhao Correct, however an anonymous user can perform actions like creating posts/comments and the `userId` would be populated. If the user deletes their cookies, they will lose that account. If the user signs in with their email address, then all of the data they created would be associated to the anonymous user's data

Of course there will be a lot of stale accounts, developers can set up cronjobs to delete accounts that havent been used in x months.

**Comment by charIeszhao** [comment]
I see... The only problem is like you said, it might end up with tons of stale accounts in the user table. And since Logto is not only an auth solution but also an identity management system, those anonymous user accounts will be hard to manage.

So maybe you can implement this layer yourself, in your own table, and only call Logto when an anonymous user fulfills their userinfo?

**Comment by AlexRMU** [comment]
See also https://supabase.com/docs/guides/auth/auth-anonymous
